File: 1/python/1.py
def twoSum(nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        indexs = [];
        dictIndex = {};
        for i in range(len(nums)):
            print('序号：%d 值: %d' % (i, nums[i]))
            if (target - nums[i]) not in dictIndex:
                dictIndex[nums[i]] = i;
            else:
                indexs = [dictIndex[target - nums[i]], i];
                break;
        return indexs;


indexs = twoSum([3,3], 6);
for i in indexs:
  print(i);

File: 1/javascript/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let results = [];
  let mapRet = new Map();
  for (let i in nums) {
    if (!mapRet.has(target - nums[i])) {
      mapRet.set(nums[i], i);
    } else {
      results = [Number(mapRet.get(target - nums[i])), Number(i)];
      break;
    }
  }
  return results;
};

function printMap(map) {
  for (let item of map) {
    console.log(item[0] + ' : ' + item[1] );
  }
}

let indexs = twoSum([2,7,11,15], 9);
for (let i in indexs) {
  console.log(indexs[i]);
}